src/server/replica.cc

```diff
@@ -588,6 +592,9 @@ error_code Replica::ConsumeDflyStream() {
         tx_data.second.block.Cancel();
       }
     }
+
+    // Reset shard states from replica state to normal.
+    SetShardStates(false);
```

Comment:
same here I think this should not be in the error handler

Reply:
I moved it into the fiber scope, because its hard to place it into the local cleanup. And as we found out below we don't actually need to revert from this state on errors